lab_3/main.py

```diff
@@ -20,7 +20,52 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
+    if not isinstance(text, str):
+        return ()
+
+    tokenized_sentence = ''
+    tokenized_sentences = []
+    underscore = ['_']
+
+    for symbol in text:
+        if symbol == 'ö':
+            text = text.replace(symbol, 'oe')
+        elif symbol == 'ü':
+            text = text.replace(symbol, 'ue')
+        elif symbol == 'ä':
+            text = text.replace(symbol, 'ae')
+        elif symbol == 'ß':
+            text = text.replace(symbol, 'ss')
+
+    for symbol in text:
+        if symbol in ['~', '`', '@', '"', '#', '№', '$', ';', '%', '^', ':', '&', '*', '(', ')', '-', '_'
+                      '=', '+', "'", '{', '[', ']', '}', '|', '\\', ':', ';', '"', "'", '<', ',', '>', '/', '1', '2',
+                      '3', '4', '5', '6', '7', '8', '9', '0']:
+            text = text.replace(symbol, '')
+
+    for symbol in text:
+        if symbol.isspace() or symbol.isalpha():
+            tokenized_sentence += symbol
+        else:
+            tokenized_sentences.append(tokenized_sentence.lower())
+            tokenized_sentence = ''
+
+    for x, tokenized_sentence in enumerate(tokenized_sentences):
+        tokenized_sentences[x] = tokenized_sentence.split()
+
+    for tokenized_sentence in tokenized_sentences:
+        for y, word in enumerate(tokenized_sentence):
+            underscore.extend(word)
```

Comment:
this variable was defined earlier - if it is used here only, consider using the local loop variable here and define it here too.